qiskit_experiments/library/characterization/cr_hamiltonian.py

```diff
@@ -0,0 +1,472 @@
+            # add cross resonance tone
+            pulse.play(
+                pulse.GaussianSquare(
+                    duration,
+                    amp=opt.amp,
+                    sigma=sigma,
+                    risefall_sigma_ratio=opt.risefall,
+                ),
+                pulse.control_channels(*self.physical_qubits)[0],
+            )
+            # add cancellation tone
+            if not np.isclose(opt.amp_t, 0.0):
+                pulse.play(
+                    pulse.GaussianSquare(
+                        duration,
+                        amp=opt.amp_t,
+                        sigma=sigma,
+                        risefall_sigma_ratio=opt.risefall,
+                    ),
+                    pulse.drive_channel(self.physical_qubits[1]),
+                )
+            else:
+                pulse.delay(duration, pulse.drive_channel(self.physical_qubits[1]))
+
+            # place holder for empty drive channels. this is necessary due to known pulse gate bug.
+            pulse.delay(duration, pulse.drive_channel(self.physical_qubits[0]))
+
+        cr_circuit.add_calibration(
+            gate=cr_gate,
+            qubits=self.physical_qubits,
+            schedule=cross_resonance,
+            params=[duration],
+        )
+
+        return cr_circuit
+
+    def _net_duration(
+        self,
+        duration: float,
+        sigma: float,
+    ) -> float:
+        """Calculate net cross resonance pulse duration considering the pulse edges.
+
+        Args:
+            duration: Parameter object representing a duration of cross resonance pulse.
+            sigma: Sigma of Gaussian edges.
+
+        Returns:
+            Net duration in units of sec.
+        """
+        opt = self.experiment_options
+
+        flat_top_width = duration - 2 * opt.risefall * sigma
+        net_edge_width = np.sqrt(2 * np.pi * sigma ** 2)
+
+        return flat_top_width + net_edge_width
+
+    def circuits(self, backend: Optional[Backend] = None) -> List[QuantumCircuit]:
+
+        opt = self.experiment_options
+        prefactor = 1.0
+        timing_constraints = getattr(backend.configuration(), "timing_constraints", dict())
+        granularity = timing_constraints.get("granularity", 1)
+
+        try:
+            dt_factor = backend.configuration().dt
+        except AttributeError as ex:
+            raise AttributeError("Backend configuration does not provide time resolution.") from ex
+
+        if opt.unit != "dt":
+            if opt.unit != "s":
+                prefactor *= apply_prefix(1.0, opt.unit)
+            prefactor /= dt_factor
+
+        # Define pulse gate of cross resonance tone
+        duration = circuit.Parameter("duration")
+        sigma_dt = prefactor * opt.sigma
+
+        template_circuits = list()
+        for control_state in (0, 1):
+            for meas_basis in ("x", "y", "z"):
+                tomo_circ = QuantumCircuit()
+                tomo_circ.add_register(2, 1)
```

Comment:
Is this different from
```
tomo_circ = QuantumCircuit(2, 1)
```
?

Reply:
Done in https://github.com/Qiskit/qiskit-experiments/pull/385/commits/c4a1391d752709bae8e86c3baf977bab808a0c73